types/src/account_address.rs

```diff
@@ -216,3 +214,31 @@ impl TryFrom<AccountAddress> for Bech32 {
         bech32::Bech32::new(LIBRA_NETWORK_ID_SHORT.into(), base32_hash).map_err(Into::into)
     }
 }
+
+impl<'de> Deserialize<'de> for AccountAddress {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            AccountAddress::try_from(s).map_err(<D::Error as serde::de::Error>::custom)
+        } else {
+            let b = <[u8; ADDRESS_LENGTH]>::deserialize(deserializer)?; //.map_err(<D::Error as serde::de::Error>::custom)?;
+            Ok(AccountAddress::new(b))
+        }
+    }
+}
+
+impl Serialize for AccountAddress {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if serializer.is_human_readable() {
+            self.to_string().serialize(serializer)
```

Comment:
I don't think that's the case: https://docs.rs/hex/0.3.1/hex/fn.encode.html

Also this would have caused lots of tests to fail and changed the syntax of configs upon running the terraform script updater.

Agree an integration test here would be good!

Reply:
Turns out our code was broken and not paying attention to the '#' formatting flag that we pass in to the Display impl, which is why it currently works as is. This doesn't really have anything to do with the hex crate but rather the LowerHex impl of AccountAddress: https://github.com/bmwill/libra/commit/0d7ec1dfa0b5008340218054e477652771b52241

Reply:
I see, I see. Looking more carefully it looks like # just ticks a flag for alternate and it would be up to the LowerHex implementation to interpret that. Perhaps we should just explicitly add 0x in the display and directly call lower_hex in the serializer. Though we really ought to have symmetry between to and from string. Your thoughts?